File: Backend/Shoply-Backend/src/main/java/com/shoply/service/OrderService.java
package com.shoply.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.shoply.entity.Order;
import com.shoply.repository.OrderRepository;

@Service
public class OrderService {

    @Autowired
    private OrderRepository orderRepository;

    @Autowired
    private EmailService emailService;

    // Get All Orders
    public List<Order> getAllOrders() {
        return orderRepository.findAll();
    }

    // Save Order + Send Email
    public Order saveOrder(Order order) {

        System.out.println("==================================");
        System.out.println("SAVE ORDER START");
        System.out.println("==================================");

        Order savedOrder = orderRepository.save(order);

        System.out.println("Order Saved Successfully");
        System.out.println("Order ID : " + savedOrder.getId());
        System.out.println("Customer : " + savedOrder.getCustomerName());
        System.out.println("Email : " + savedOrder.getEmail());

        try {

            System.out.println("Sending Email...");

            emailService.sendOrderConfirmation(
                    savedOrder.getEmail(),
                    savedOrder.getCustomerName(),
                    savedOrder.getId(),
                    savedOrder.getAmount(),
                    savedOrder.getAddress()
            );

            System.out.println("==================================");
            System.out.println("EMAIL SENT SUCCESSFULLY");
            System.out.println("==================================");

        } catch (Exception e) {

            System.out.println("==================================");
            System.out.println("EMAIL SENDING FAILED");
            System.out.println("==================================");

            e.printStackTrace();
        }

        return savedOrder;
    }

    // Get Order By Id
    public Order getOrderById(Long id) {
        return orderRepository.findById(id).orElse(null);
    }

    // Delete Order
    public void deleteOrder(Long id) {
        orderRepository.deleteById(id);
    }

    // Update Order
    public Order updateOrder(Long id, Order order) {
        order.setId(id);
        return orderRepository.save(order);
    }
}

File: Frontend/shopping/src/services/orderService.js
import axios from "axios";

const API_URL = "http://localhost:8080/orders";

export const getOrders = async () => {
  const response = await axios.get(API_URL);
  return response.data;
};

export const addOrder = async (order) => {
  const response = await axios.post(API_URL, order);
  return response.data;
};

export const getOrderById = async (id) => {
  const response = await axios.get(`${API_URL}/${id}`);
  return response.data;
};

export const updateOrder = async (id, order) => {
  const response = await axios.put(`${API_URL}/${id}`, order);
  return response.data;
};

export const deleteOrder = async (id) => {
  await axios.delete(`${API_URL}/${id}`);
};